src/twister2/filter/quarantine_filter.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import pytest
+import logging
+
+from pathlib import Path
+from yaml import safe_load
+from marshmallow import Schema, fields, ValidationError  # validates
+from dataclasses import dataclass, field
+
+from twister2.filter.filter_interface import FilterInterface
+from twister2.report.helper import get_test_name, get_item_platform
+from twister2.exceptions import TwisterConfigurationException
+
+logger = logging.getLogger(__name__)
+
+
+class QuarantineFilter(FilterInterface):
+    """Filter tests under quarantine."""
+
+    def __init__(self, config: pytest.Config) -> None:
+        """
+        :param config: pytest configuration
+        """
+        super().__init__(config)
+        self.quarantine_verify = config.getoption('--quarantine-verify')
+        self.quarantine = QuarantineData()
+        for quarantine_file in config.getoption('--quarantine-list'):
+            self.quarantine.extend(QuarantineData.load_data_from_yaml(quarantine_file))
+
+    def filter(self, item: pytest.Item) -> bool:
+        """
+        Check if test should be deselected
+
+        :param item: pytest test item
+        :return: True if test should be deselected
+        """
+        if qelem := self.quarantine.get_matched_quarantine(item):
+            if not self.quarantine_verify:
+                logger.debug(f"Skipped test {get_test_name(item)} - quarantine reason: {qelem.comment}")
+                # only mark test to be skipped, this test still will be listed in test plan,
+                # but will not be executed
+                item.add_marker(pytest.mark.skip(f'Quarantine: {qelem.comment}'))
```

Comment:
It works as expected: test is only marked to be skipped, but hook returns False, it means that test is not deselected. But after   offline discussion I agree that it would be better to not use filter plugin to mark tests to be under quarantine, but to use separate plugin uses `pytest_collection_modifyitems` to mark quarantined tests. I will try to implement such a solution.